src/services/lease-service/index.ts

```diff
@@ -304,6 +304,50 @@ export const routes = (router: KoaRouter) => {
     ctx.body = { content: result.data, ...metadata }
   })
 
+  /**
+   * @swagger
+   * /offers/listing-id/{listingId}:
+   *   get:
+   *     summary: Gets active offer for a specific listing
+   *     description: Get an offer for a listing.
+   *     tags: [Offer]
+   *     parameters:
+   *       - in: path
+   *         name: listingId
+   *         required: true
+   *         schema:
+   *           type: integer
+   *         description: The unique ID of the listing.
+   *     responses:
+   *       200:
+   *         description: The active offer.
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: object
+   *       500:
+   *         description: Internal server error.
+   */
+  router.get('/offers/listing-id/:listingId/active', async (ctx) => {
+    const metadata = generateRouteMetadata(ctx)
+    const result = await leasingAdapter.getActiveOfferByListingId(
+      Number.parseInt(ctx.params.listingId)
+    )
+
+    if (!result.ok) {
+      if (result.err === 'not-found') {
+        ctx.status = 404
+        return
```

Comment:
In my opinion there should always be a body with {error: string} regardless of the status code returned. It will make translation of errors so much easier in the frontend.

Reply:
https://github.com/Bostads-AB-Mimer/onecore-core/pull/188/commits/8b60bf3995c39a5f240a7c521c53b3a474cb634b